### DISCUSSION TITLE
Lifecycle Hooks - Executing on Load of Extension

### DISCUSSION THREAD

**Discussion by willbonde** [zephir-lang/zephir#2289] [Q&A] [answered]
I had a question regarding Lifecycle Hooks, probably out of my own confusion. I'd like to author an extension that I'd like to run some code when the PHP interpreter is loaded (well, I guess that is to say when my extension is loaded). Can that be done with the Lifecycle Hooks? If so, can I use code authored in Zephir, or just C? Thanks!

**Comment by AlexNDRmac** [comment, marked as answer]
Hi @willbonde. The answer - yes... in generally. You should use RINIT hook for this. With this hook you will have a Request and the ability to work with memory, zend objects, etc. You can read more about the PHP Lifecycle and Hooks [here](https://www.phpinternalsbook.com/php7/extensions_design/php_lifecycle.html#the-php-extensions-hooks)

**Reply by willbonde** [reply]
Wonderful thank you! Will read up on that today :)

**Comment by praswicaksono** [comment]
Could be user if someone able to create simple example about life cycle hook

**Reply by danhunsaker** [reply]
I included a very simple example in the Zephir [Lifecycle Hooks](https://docs.zephir-lang.com/latest/en/lifecycle) documentation, but if it needs to be expanded, I can take a look!
